packages/core/src/hooks/useContractCall.ts

```diff
@@ -2,15 +2,17 @@ import { Interface } from '@ethersproject/abi'
 import { useMemo } from 'react'
 import { Falsy } from '../model/types'
 import { useChainCalls } from './useChainCalls'
-import { RawCall } from '../providers/chainState/callsReducer'
+import { RawCall, useNetwork } from '../providers'
+import { ChainId } from '../constants'
+import { QueryParams } from '../constants/type/QueryParams'
 
 function warnOnInvalidContractCall(call: ContractCall | Falsy) {
   console.warn(
     `Invalid contract call: address=${call && call.address} method=${call && call.method} args=${call && call.args}`
   )
 }
 
-function encodeCallData(call: ContractCall | Falsy): RawCall | Falsy {
+function encodeCallData(call: ContractCall | Falsy, chainId: ChainId): RawCall | Falsy {
```

Comment:
Shouldn't we use option object here?

Reply:
options object is for public apis, and this function is internal and only cares about chainId